src/avram/type.cr

```diff
@@ -14,8 +14,7 @@ module Avram::Type
   def parse(values : Array(String))
     casts = values.map { |value| parse(value) }
     if casts.all?(&.is_a?(SuccessfulCast))
-      values = casts.map { |c| c.as(SuccessfulCast).value }
-      parse(values)
+      casts
```

Comment:
I think the original code fixed a bug in the Crystal compiler that now seems to be fixed :D